src/index.ts

```diff
@@ -25,6 +25,22 @@ export interface Collection<K, V> extends Map<K, V> {
 export class Collection<K, V> extends Map<K, V> {
 	public static readonly default: typeof Collection = Collection;
 
+	/**
+	 * Obtains an element if the key exists, otherwise sets and returns the value provided by the default value generator.
+	 *
+	 * @param key The key to get from if it exists or set otherwise
+	 * @param defaultValueGenerator A function that returns the ensured value
+	 *
+	 * @example
+	 * collection.ensure(guildId, () => defaultGuildConfig);
+	 */
+	public ensure(key: K, defaultValueGenerator: (key: K, collection: this) => V): V | undefined {
+		if (this.has(key)) return this.get(key);
+		const defaultValue = defaultValueGenerator(key, this);
+		this.set(key, defaultValue);
+		return defaultValue;
```

Comment:
This could theoretically all be moved into one line, but don't apply this suggestion without more feedback from the maintainers 
```suggestion
		return this.get(key) ?? this.set(key, defaultValueGenerator(key, this)).get(key)
```

Reply:
Please don't. This is a lot of unnecessary overhead.

Reply:
Not sure where the PR's description mentions this but I figured someone would say that. Still it's quite small and easy to ready but I understand your concerns

Reply:
This line:
![image](https://user-images.githubusercontent.com/24852502/143323627-54674bc5-b037-40eb-bb55-339ee959ce7d.png)